packages/mc-scripts/src/commands/login.ts

```diff
@@ -18,27 +63,111 @@ async function run() {
     return;
   }
 
-  console.log(`Enter the login credentials:`);
+  if (shouldUseExperimentalIdentityAuthFlow) {
+    const open = await import('open');
 
-  const { email } = await prompts({
-    type: 'text',
-    name: 'email',
-    message: 'Email',
-  });
-  const { password } = await prompts({
-    type: 'invisible',
-    name: 'password',
-    message: 'Password (hidden)',
-  });
+    const state = generateRandomHash();
+    const nonce = generateRandomHash();
 
-  if (!email || !password) {
-    throw new Error(`Missing email or password values. Aborting.`);
-  }
+    const authUrl = new URL('/login/authorize', mcApiUrl);
+    authUrl.searchParams.set('response_type', 'id_token');
+    authUrl.searchParams.set('response_mode', 'form_post');
+    authUrl.searchParams.set('client_id', '__local:@@cli@@');
+    authUrl.searchParams.set(
+      'scope',
+      [
+        'openid',
+        // 'project_key:??',
+      ].join(' ')
+    );
+    authUrl.searchParams.set('state', state);
+    authUrl.searchParams.set('nonce', nonce);
+
+    const server = http.createServer(async (request, response) => {
+      try {
+        if (
+          request.url?.includes('/oidc/callback') &&
+          request.method === 'POST' &&
+          request.headers['content-type'] ===
+            'application/x-www-form-urlencoded'
+        ) {
+          const body = await parseRequestBody(request);
+          const authParams = decode<AuthorizeCallbackFragments>(body);
```

Comment:
The request body contains the token (because of `response_mode: form_post`), so we parse it. The `decode` function simply transforms the data from url-form-encoded to JSON.